apps/next/components/Details/index.tsx

```diff
@@ -29,7 +29,7 @@ export default function Details({ result }: { result: string }) {
 
   // State variable that changed inside Wagmi hooks
   const [prepared, setPrepared] = useState<boolean>(false)
-  const [isAvailable, setIsAvailable] = useState<boolean>(true)
+  const [isAvailable, setIsAvailable] = useState<boolean>()
```

Comment:
Why is `isAvailable` no longer `true` by default? Also wouldn't every domain default to `false` until availability has been confirmed with a contract read?

Reply:
Same here. For the purpose of animation, I believe it is better to display a loading state before showing the actual value. From a user experience standpoint, it is not visually appealing to display that a domain is unavailable and then suddenly show that it is actually available. Additionally, if there is a situation with a poor internet connection, the user may see that the domain is unavailable for several seconds because the default value is displayed instead of the actual value. By handling this situation with a loading screen, I believe it provides a better experience for the user. Another, and even better, solution is to use an enum, as you suggested, to handle both situations we discussed. For example: 
```const State = {
  Registered: true,
  Unregistered: false,
  Unchecked: undefined,
} as const;```

Reply:
I would agree that having loading states before actual information has been read or loaded is better for the user experience. Thank you again for the thoughtful and detailed explanation.